tests/README.md

```diff
@@ -0,0 +1,156 @@
+The idea is to check that repeated execution of the examples produce same serial output. An example produces serial output when code contains ```printf``` statements. Serial output may change for ligitimate reasons like use of random data or data and time stamps. Thats why the log is converted to a template. This means that either the text/lines that differ in every execution are removed or they are converted into regular expressions. See the example below:
+
+```
+
+							      >	Using Ethernet LWIP
+								
+							      >	Client IP Address is 10.2.203.139
+								
+							      >	Connecting with developer.mbed.org
+								
+Starting the TLS handshake...								Starting the TLS handshake...
+								
+							      >	TLS connection to developer.mbed.org established
+								
+Server certificate:								Server certificate:
+								
+							      >	
+								    cert. version     : 3
+							      >	
+								    serial number     : 11:21:B8:47:9B:21:6C:B1:C6:AF:BC:5D:0
+							      >	
+								    issuer name       : C=BE, O=GlobalSign nv-sa, CN=GlobalSi
+							      >	
+								    subject name      : C=GB, ST=Cambridgeshire, L=Cambridge,
+							      >	
+								    issued  on        : 2016-03-03 12:26:08
+							      >	
+								    expires on        : 2017-04-05 10:31:02
+							      >	
+								    signed using      : RSA with SHA-256
+							      >	
+								    RSA key size      : 2048 bits
+							      >	
+								    basic constraints : CA=false
+							      >	
+								    subject alt name  : *.mbed.com, mbed.org, *.mbed.org, mbe
+							      >	
+								    key usage         : Digital Signature, Key Encipherment
+							      >	
+								    ext key usage     : TLS Web Server Authentication, TLS We
+
+Certificate verification passed								Certificate verification passed
+								
+
+								
+								
+							      >	HTTPS: Received 439 chars from server
+								
+							      >	HTTPS: Received 200 OK status ... [OK]
+								
+HTTPS: Received 'Hello world!' status ... [OK]								HTTPS: Received 'Hello world!' status ... [OK]
+								
+HTTPS: Received message:								HTTPS: Received message:
+								
+
+								
+								
+							      >	HTTP/1.1 200 OK
+								
+							      >	Server: nginx/1.7.10
+								
+							      >	Date: Thu, 01 Dec 2016 13:56:32 GMT
+								
+							      >	Content-Type: text/plain
+								
+							      >	Content-Length: 14
+								
+							      >	Connection: keep-alive
+								
+							      >	Last-Modified: Fri, 27 Jul 2012 13:30:34 GMT
+								
+							      >	Accept-Ranges: bytes
+								
+							      >	Cache-Control: max-age=36000
+								
+							      >	Expires: Thu, 01 Dec 2016 23:56:32 GMT
+								
+							      >	X-Upstream-L3: 172.17.0.3:80
+								
+							      >	X-Upstream-L2: developer-sjc-indigo-2-nginx
+								
+							      >	Strict-Transport-Security: max-age=31536000; includeSubdomain
+								
+
+								
+								
+Hello world!								Hello world!
+
+```
```

Comment:
I would leave this example as it illustrates the one way of making a template log that is by removing random data.